### PR TITLE
separate lightnode(combine wallet with txgen) from normal node; refactoring peer discovery logic

### PR DESCRIPTION
**Pull request by chaosma**

### CODE REVIEW

File: cmd/client/txgen/main.go [hunk 1]
```diff
@@ -152,10 +152,6 @@ func main() {
 	clientNode.ServiceManagerSetup()
 	clientNode.RunServices()
 
-	// Start the client server to listen to leader's message
```
Comment (LeoHChen) [inline cmd/client/txgen/main.go]: why remove those?

File: cmd/client/wallet/lib/lib.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package lib
 
 import (
 	"fmt"
-	"time"
+	//	"time"
```
Comment (LeoHChen) [inline cmd/client/wallet/lib/lib.go]: remove it if it is not needed.

File: go.mod [hunk 3]
```diff
@@ -1,106 +1,38 @@
 module github.com/harmony-one/harmony
```
Comment (LeoHChen) [inline go.mod]: why change go.mod again?

File: node/node.go [hunk 4]
```diff
@@ -407,76 +407,33 @@ func (node *Node) RemovePeersHandler() {
 	}
 }
 
-func (node *Node) initWalletNodeConfiguration() (service.NodeConfig, chan p2p.Peer) {
+// isBeacon = true if the node is beacon node
+// isClient = true if the node light client(txgen,wallet)
+func (node *Node) initNodeConfiguration(isBeacon bool, isClient bool) (service.NodeConfig, chan p2p.Peer) {
```
Comment (LeoHChen) [inline node/node.go]: we can actually use the global nodeconfig now, instead of passing the parameters here.

File: node/service_setup.go [hunk 5]
```diff
@@ -48,7 +48,7 @@ func (node *Node) setupForShardValidator() {
 }
 
 func (node *Node) setupForBeaconLeader() {
-	nodeConfig, chanPeer := node.initBeaconNodeConfiguration()
+	nodeConfig, chanPeer := node.initNodeConfiguration(true, false)
```
Comment (LeoHChen) [inline node/service_setup.go]: let's use the global configuration nodeconfig

File: api/client/service/proto/client.pb.go [hunk 6]
```diff
@@ -20,7 +21,7 @@ var _ = math.Inf
 // is compatible with the proto package it is being compiled against.
 // A compilation error at this line likely means your copy of the
 // proto package needs to be updated.
-const _ = proto.ProtoPackageIsVersion3 // please upgrade the proto package
+const _ = proto.ProtoPackageIsVersion2 // please upgrade the proto package
```
[the remark this answers is not in the document]
Comment (chaosma) [reply api/client/service/proto/client.pb.go]: this is automatically generated, don't know why it changed

File: cmd/client/wallet/lib/lib.go (hunk 2 above)
Comment (chaosma) [reply cmd/client/wallet/lib/lib.go]: the wallet part is not fixed yet. I am currently switching between hard code peer info with p2p discovery

File: go.mod (hunk 3 above)
Comment (chaosma) [reply go.mod]: current go.mo not work on my machine, I have to generate a new go.mo

File: node/node.go (hunk 4 above)
Comment (chaosma) [reply node/node.go]: Currently we are using NodeConfig and ConfigType, let's remove NodeConfig in another pr

File: node/service_setup.go (hunk 5 above)
Comment (chaosma) [reply node/service_setup.go]: let's do it in a separate pr

File: cmd/client/txgen/main.go (hunk 1 above)
Comment (chaosma) [reply cmd/client/txgen/main.go]: added back

File: api/client/service/proto/client.pb.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (chaosma) [reply api/client/service/proto/client.pb.go]: I used the newer version, now it's fixed

File: cmd/client/wallet/lib/lib.go [hunk 7]
```diff
@@ -60,7 +59,8 @@ func SubmitTransaction(tx *types.Transaction, walletNode *node.Node, shardID uin
 }
 
 func getBootNodes() []ma.Multiaddr {
-	addrStrings := []string{"/ip4/54.213.43.194/tcp/9874/p2p/QmdZ7ccH4A6jr6kKzdhx2ttfKz6AEeTpwMChysUL1QLbYX"}
+	addrStrings := []string{"/ip4/127.0.0.1/tcp/19876/p2p/QmbTLMb9C8dmjrDYoiJb2mayXspcURkNB4ARxgsoA5Aoe3"}
```
Comment (LeoHChen) [inline cmd/client/wallet/lib/lib.go]: no need to check-in the change of this line

File: node/node_test.go [hunk 8]
```diff
@@ -166,8 +166,10 @@ func sendPingMessage(node *Node, leader p2p.Peer) {
 		ConsensusPubKey: pubKey1,
 	}
 
-	ping1 := proto_discovery.NewPingMessage(p1)
+	ping1 := proto_discovery.NewPingMessage(p1, true)
+	ping2 := proto_discovery.NewPingMessage(p1, false)
 	_ = ping1.ConstructPingMessage()
+	_ = ping2.ConstructPingMessage()
```
Comment (LeoHChen) [inline node/node_test.go]: any meaningful test or check ?

Comment (chaosma) [reply node/node_test.go]: check true/false combination of different initialization

**Pull request merged**